sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+String contentType = "type";
+String version = Constants.HeaderConstants.TARGET_STORAGE_VERSION;
+
+// build the token
+String sasToken = blobClient.generateSAS(identifier, permissions, expiryTime, startTime, version, sasProtocol,
+            ipRange, cacheControl, contentDisposition, contentEncoding, contentLanguage, contentType);
+```
+
+## Build Client
+In v10/v11, core classes were built with constructors, taking a `URL` and an already constructed `HttpPipeline`.
+```java
+final BlockBlobURL blobURL = new BlockBlobURL(
+    new URL("https://" + accountName + ".blob.core.windows.net/mycontainer/myimage.jpg"), 
+    StorageURL.createPipeline(myCredential, new PipelineOptions()));
+```
+
+In v12, we have moved to a builder pattern. To replicate the above v10/v11 snippit:
+```java
+final BlockBlobClient blobClient = new BlobClientBuilder()
+    .endpoint("https://" + accountName + ".blob.core.windows.net")
+    .containerName("mycontainer")
+    .blobName("myimage.jpg")
+    .credential(myCredential)
+    .buildBlockBlobClient(); // we can also choose PageBlobClient, AppendBlobClient, and BlobClient
+```
+These builders can also build sync and async clients interchangably. Just select the correct "build" method. Additionally, if you already have the full URL, you can just specify that as the endpoint instead of naming container and blob individually, similar to v10/v11.
+```java
+ final BlockBlobClient blobClient = new BlobClientBuilder()
+    .endpoint("https://" + accountName + ".blob.core.windows.net/mycontainer/myimage.jpg")
+    .credential(myCredential)
+    .buildBlockBlobAsyncClient(); // we choose the "async" method this time
+``` 
+
+## Credentials
+
+### Connection string 
+v10/v11 did not support connection strings. In v12, our builders accept connection strings, no calls to `endpoint()` or `credential()` required. However, connection strings generally only point at the account, so you WILL need to specify container and blob names as necessary.
+```java
+ContainerClient client = new ContainerClientBuilder()
+    .connectionstring("${connectionstring}")
+    .containerName("mycontainer")
+    .buildBlockBlobClient();
+```
+
+### Shared key credential
+Shared key credentials can be used directly to authenticate your client. You need to fetch the `${accountName}` and `${accountKey}` from Azure Portal.
+Learn More from [README](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob/README.md)
+
+In v10/v11, you used the credential to build your pipeline.
+```java
+// Initialize the credentials
+SharedKeyCredentials credential = new SharedKeyCredentials("${accountName}", "${accountKey}");
+// How service object take in the credentials.
+BlockBlobURL blockBlobURL = new BlockBlobURL(new URL("<your-storage-blob-url>"), StorageURL.createPipeline(credential, new PipelineOptions()));
+```
+
+Since the builder in v12 manages pipeline generation, you can just hand the `SharedKeyCredential` to the builder.
```

Comment:
I think we should also mention that what was in PipelineOptions is now flattened to just be on the builder. e.g. RequestRetryOptions